ansible_runner/__main__.py

```diff
@@ -793,18 +792,13 @@ def main(sys_args=None):
             print(safe_dump(info, default_flow_style=True))
             parser.exit(0)
 
-        cleanup_data_dir = True
-        if vargs.get('private_data_dir'):
-            if os.path.exists(vargs.get('private_data_dir')):
-                if vargs.get('delete_directory', False):
-                    shutil.rmtree(vargs['private_data_dir'])
-                else:
-                    cleanup_data_dir = False
-        else:
+        if vargs.get('private_data_dir') and vargs.get('delete_directory', False):
+            cleanup_folder(vargs['private_data_dir'])
+            register_for_cleanup(vargs['private_data_dir'])
+        elif not vargs.get('private_data_dir'):
```

Comment:
For clarity, store the values. This would be a great place to use an assignment expression (`:=`) but I don't think we're ready to make Python 3.8 a hard requirement yet.
```suggestion
        private_data_dir = vargs.get('private_data_dir')
        delete_directory = vargs.get('delete_directory', False)
        if private_data_dir and delete_directory:
            shutil.rmtree(private_data_dir, True)
            register_for_cleanup(private_data_dir)removed?
        elif private_data_dir is None:
```

Reply:
I'm not sure about the call to `register_for_cleanup()` after removing the directory. Does it get recreated and this is to ensure the directory is removed before it is later created? It's probably worth putting a comment in here explaining the reason behind the code.

Reply:
These two lines track with what was said in the docs. If the directory exists before the run, it deletes it, and it also deletes it after the run. So, deletion before and after. The run will _always_ create the directory, because ansible-runner works via a private data dir.